integrations/tensorflow/bindings/python/pyiree/tf/support/module_utils.py

```diff
@@ -40,108 +40,64 @@ def _running_bazel_test() -> bool:
-        compiler.Context.default_crash_reproducer_path = None
-      raise
-    return results
-
-  return decorator
-
-
-def _incrementally_lower_compiler_module(
-    compiler_module: compiler.Module,
-    backend_info: "BackendInfo",
-    artifacts_dir: str,
-) -> Tuple[compiler.binding.OpaqueBlob, Union[str, None]]:
-  """Lowers a MLIR compiler module incrementally and saves its outputs.
-
-  If artifacts_dir is provided then the following artifacts will be saved:
+  When artifacts_dir is set, writes:
     tf_input.mlir:
       MLIR for the module in TF's input dialect.
     iree_input.mlir:
       The MLIR above translated to IREE via compiler.TF_IMPORT_PASS_PIPELINE.
     backend_id/compiled.vmfb:
       A VM FlatBuffer compiled to the target backend from the IREE MLIR above.
+    `artifacts_dir/reproducer__{backend}.mlir` in the case of a crash.
 
   Args:
-    compiler_module: A compiler.Module to lower.
-    backend_info: BackendInfo with the details for lowering compiler_module to
-      IREE.
-    artifacts_dir: An optional string pointing to where compilation artifacts
-      should be saved. No compilation artifacts will be saved if this is not
-      provided.
+    artifacts_dir: The artifacts directory.
+    backend_id: The backend id (for artifacts that are backend dependent).
+    needs_temp_saved_model_dir: Whether a temporary 'saved_model_dir' directory
+      needs to be set.
+  Returns:
+    A dict of output kwargs.
   """
-  if artifacts_dir is not None:
-    os.makedirs(artifacts_dir, exist_ok=True)
-    tf_mlir_path = os.path.join(artifacts_dir, "tf_input.mlir")
-    logging.info("Saving raw TF input MLIR to: %s", tf_mlir_path)
-    with open(tf_mlir_path, "w") as f:
-      f.write(compiler_module.to_asm())
+  kwargs = {}
+  backend_dir = os.path.join(artifacts_dir, backend_id)
+  os.makedirs(backend_dir, exist_ok=True)
+  kwargs["output_file"] = os.path.join(backend_dir, "compiled.vmfb")
+  if needs_temp_saved_model_dir:
+    kwargs["saved_model_dir"] = os.path.join(artifacts_dir,
+                                             "tfmodule.saved_model")
+  kwargs["save_temp_tf_input"] = os.path.join(artifacts_dir, "tf_input.mlir")
+  kwargs["save_temp_iree_input"] = os.path.join(artifacts_dir,
+                                                "iree_input.mlir")
 
-  # Manually run the passes that tf_module_to_compiler_module usually would.
-  compiler_module.run_pass_pipeline(compiler.TF_IMPORT_PASS_PIPELINE)
-
-  if artifacts_dir is not None:
-    iree_mlir_path = os.path.join(artifacts_dir, "iree_input.mlir")
-    logging.info("Saving IREE input MLIR to: %s", iree_mlir_path)
-    with open(iree_mlir_path, "w") as f:
-      f.write(compiler_module.to_asm())
-
-  compiled_module = compiler_module.compile(
-      target_backends=backend_info.compiler_targets)
+  # Avoid the crash reproducer under tests or if the flag is false.
+  # Developers can run tests outside of the test runner (e.g. `bazel run`) to
```

Comment:
We can revisit this later. If never run under bazel, this might not be an issue.